client/src/app/containers/Layout/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+
+import Divider from '@material-ui/core/Divider';
+import Drawer from '@material-ui/core/Drawer';
+import IconButton from '@material-ui/core/IconButton';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
+import clsx from 'clsx';
+
+import { useAuth } from '../../../auth/useAuth';
+import Navbar from '../../components/Navbar';
+import Alert from '../Alert';
+import LoggedInDrawer from '../LeftDrawer/LoggedInDrawer';
+import LoggedOutDrawer from '../LeftDrawer/LoggedOutDrawer';
+
+const drawerWidth = 240;
+
+const useStyles = makeStyles(theme => ({
+    root: {
+        display: 'flex',
+    },
+    drawer: {
+        width: drawerWidth,
+        flexShrink: 0,
+    },
+    drawerPaper: {
+        width: drawerWidth,
+    },
+    drawerHeader: {
+        display: 'flex',
+        alignItems: 'center',
+        padding: theme.spacing(0, 1),
+        // necessary for content to be below app bar
+        ...theme.mixins.toolbar,
+        justifyContent: 'flex-end',
+    },
+    content: {
+        flexGrow: 1,
+        padding: theme.spacing(3),
+        transition: theme.transitions.create('margin', {
+            easing: theme.transitions.easing.sharp,
+            duration: theme.transitions.duration.leavingScreen,
+        }),
+        marginLeft: -drawerWidth,
+    },
+    contentShift: {
+        transition: theme.transitions.create('margin', {
+            easing: theme.transitions.easing.easeOut,
+            duration: theme.transitions.duration.enteringScreen,
+        }),
+        marginLeft: 0,
+    },
+}));
+
+const Layout: React.FunctionComponent = ({ children }) => {
+    const { isLoggedIn } = useAuth();
+
+    const [drawerOpen, setDrawerOpen] = useState(false);
+    const handleDrawerOpen = () => {
+        setDrawerOpen(true);
+    };
+    const handleDrawerClose = () => {
+        setDrawerOpen(false);
+    };
+
+    const classes = useStyles();
+
+    return (
+        <>
+            <Navbar handleDrawerOpen={handleDrawerOpen} />
+            <div className={classes.root}>
+                <Drawer
+                    className={classes.drawer}
+                    variant='persistent'
+                    anchor='left'
+                    open={drawerOpen}
+                    classes={{
+                        paper: classes.drawerPaper,
+                    }}
+                >
+                    <div className={classes.drawerHeader}>
+                        <IconButton onClick={handleDrawerClose}>
+                            <ChevronLeftIcon />
+                        </IconButton>
+                    </div>
+                    <Divider />
+                    {isLoggedIn ? <LoggedInDrawer /> : <LoggedOutDrawer />}
+                </Drawer>
+                <main
```

Comment:
:mag: what is `main`?

Reply:
Not gonna lie. I copied this from Material when we were working on this a while back. But: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main